File: 1586-longest-subarray-of-1s-after-deleting-one-element/longest-subarray-of-1s-after-deleting-one-element.java
class Solution {
    public int longestSubarray(int[] nums) {
        int n = nums.length;
        int i = 0;

        int delete = 0;
        int maxCount = 0;

        for(int j = 0; j < n; j++) {
            if(nums[j] == 0) {
                delete++;
            }

            while(delete > 1) {
                if(nums[i] == 0 ) {
                    delete--;
                }
                i++;
            }

            maxCount = Math.max(maxCount, j - i);
        }

        return maxCount;

        
    }
}

File: 1586-longest-subarray-of-1s-after-deleting-one-element/longest-subarray-of-1s-after-deleting-one-element.py
class Solution(object):
    def longestSubarray(self, nums):
        """
        :type nums: List[int]
        :rtype: int
        """
        l = 0
        zero_count = 0
        ans = 0

        for r in range(len(nums)):
            if nums[r] == 0:
                zero_count += 1

            while zero_count > 1:  # shrink window until valid
                if nums[l] == 0:
                    zero_count -= 1
                l += 1

            # subtract 1 because we must delete one element
            ans = max(ans, r - l)

        return ans
